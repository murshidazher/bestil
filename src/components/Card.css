img {
  width:130px; 
  height:190px;
  /* display: inline-block;
  border-width: 0;
  box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.1), 0 9px 20px 0 rgba(0, 0, 0, 0.25);
  overflow: hidden;
  transition: box-shadow .3s linear; */

  
}




.cover:before {
  content: '';
  width: 130px;
  height: 190px;
  background: linear-gradient(to right, rgba(255, 255, 255, 0.2) 0%,rgba(255, 255, 255, 0.1) 100%);
  position: absolute;
  /* left: 13px; */
  /* top: 9px; */


}

.cover {
  text-align: center;
  margin-bottom: 16px;
  position: relative;
}
.cover::after {
  content: '';
    display: block;
    position: absolute;
    width: 8px;
    height: 190px;
    left: 31px;
    bottom: 0;
    border-radius: 2px 0 0 2px;
    background: linear-gradient(to right, rgba(255, 255, 255, 0) 0%, rgba(255, 255, 255, 0.1) 20%, rgba(255, 255, 255, 0.1) 45%, rgba(0, 0, 0, 0.15) 49%, rgba(0, 0, 0, 0.15) 84%, rgba(0, 0, 0, 0) 100%);
    background-clip: border-box;
    filter: blur(2.4px);
    opacity: 0.3;
}

.cover__img {
  border-top-right-radius: 4px;
  border-bottom-right-radius: 4px;
  border-top-left-radius: 2px;
  border-bottom-left-radius: 2px;
  box-shadow: 0 0px 9px 0px #a0a0a04d;

}

.bdr--right {
  border-left: 1px solid rgba(100, 105, 106, 0.3);
  height: 30px;
}

.tc {
  text-align: center;
  
  
}

.dib {
  display: inline-block;
  vertical-align: middle;
}


.small {
  font-size: 10px;
  margin-right: 20px;
  text-decoration: line-through;
  letter-spacing: .05em;
  margin-left: 16px;
  color: rgb(57, 57, 58);
}

.big {
  font-size: 16px;
  padding-left: 20px;
  color: #B34CF3;
}

.book {
  width: 190px;
  height: 340px;
  padding-top: 40px;
  padding-right: 8px;
}

.italics {
  font-style: italic;
  letter-spacing: .05em;
}


.book+.detail-view {
  display: none;
  -moz-osx-font-smoothing: grayscale;
  backface-visibility: hidden;
  /* transform: translateZ(.5); */
  opacity: 0.2;
  transition: opacity 1s ease-out;
  left: 200px;
  position: absolute;
}
/* .book:hover +.detail-view {
  display: block; 
  width: 210px;
  height: 280px;
  background:rgb(255, 255, 255);
  border:1px solid #f6f6f6;
  box-shadow: 0px 2px 5px 0px rgba(179,179,179,.15);
  z-index: 9999;
  opacity: 1;
  transition: opacity 1s ease-in-out;

} */

/* .detail-view{display:inline-block; margin:0 0 0 0;  box-sizing:border-box; } */
.detail-view::before{width: 0; 
  height: 0; 
  border-left: 5px solid transparent;
  border-right: 5px solid transparent;
  
  border-bottom: 5px solid black;}

  .card {
    display: flex;
    position: relative;
  }

  .cart {
    height: 32px;
    background-color: #603B32;
  }

  .details {
    height: 200px;
    padding: 20px;
  }

  p {
    margin: 0;
    font-size: 13px;
    letter-spacing: .02em;
    line-height: 1.4;
  }

  .title {
    font-size: 16px;
    max-height: 32px;
    text-overflow: ellipsis;

    /* Required for text-overflow */
    white-space: nowrap;
    overflow: hidden;
    margin-bottom: 8px;
  }

  .author {
    font-style: italic;
    font-size: 12px;
    margin-bottom: 16px;
    max-height: 32px;
    text-overflow: ellipsis;
    white-space: nowrap;
    overflow: hidden;
  }

  .desc {
    max-height: 32px;
    text-overflow: ellipsis;

    /* Required for text-overflow */
    white-space: pre-line;
    overflow: hidden;
    margin-bottom: 24px;
  }

  .mt2 {
    padding-top: 10px;
    /* padding-bottom: 8px; */
  }

  .grey {
    color: #f6f6f6;
  }

  .bold {
    font-weight: bold;
  }


  .status {
    text-align: center;
    border-radius: .25em;
    padding-top: 0.25em;
    padding-bottom: 0.25em;
    padding-right: .5em;
    padding-left: .5em;
    font-size: .8em;
  }

  .available {
    background-color: #dff7eb;
    color: #48b271;
  }
  .nostock {
    background-color: #fddedd;
    color: #e83131;
  }

  .link {
    text-decoration: underline;
    font-size: 14px;
  }

  .mt {
    /* margin-top: 1rem; */
  }

