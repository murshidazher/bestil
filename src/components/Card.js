import React from 'react';
import './Card.css';

const Card = (props) => {
  return (

    <div className="dib mt">   
      <div className="card">
        <div className="book">
          <div className="cover">
            <img className="cover__img" src={props.cover}  alt='cover'/>
          </div>
      
          <div className="tc">
            <div className="title">{props.name}</div>
            <div className="author">{props.author}</div>
            <div className="small dib"><span>$ {props.pbd.toFixed(2)}</span></div>
            <div className="dib bdr--right"></div>
            <div className="big dib italics"><span>$ {props.pad.toFixed(2)}</span></div>
            <div className="mt2">
            <span className={"status " + 
            ((props.status.toLowerCase() === 'available')? 'available' : 'nostock')}>{props.status}</span>
          </div>
          </div>
        </div>
      </div>
    </div>
  );
}

export default Card;
