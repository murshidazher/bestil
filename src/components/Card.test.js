import React from 'react';
import { shallow } from 'enzyme';
import Card from './Card';

it('except to render Card component', () => {
    // this states that if we render something and length returns 1
    // means it rendered
    const mockBook = {
        name: 'Book of John Doe',
        author: 'John Doe',
        pbd: '26.85',
        pad: '14.58',
        status: 'Available'
    };

    expect(shallow(<Card name={mockBook.name} author={mockBook.author} pbd={mockBook.pbd} pad={mockBook.pad} status={mockBook.status} />)).toMatchSnapshot();
})

it('except to render Card component with soldout', () => {
    // this states that if we render something and length returns 1
    // means it rendered
    const mockBook = {
        name: 'Book of John Doe',
        author: 'John Doe',
        pbd: '26.85',
        pad: '14.58',
        status: 'Sold Out'
    };

    expect(shallow(<Card name={mockBook.name} author={mockBook.author} pbd={mockBook.pbd} pad={mockBook.pad} status={mockBook.status} />)).toMatchSnapshot();
})

