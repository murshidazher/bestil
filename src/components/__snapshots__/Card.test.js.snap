// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`except to render Card component 1`] = `ShallowWrapper {}`;

exports[`except to render Card component with soldout 1`] = `ShallowWrapper {}`;
